README


This archive contains the distribution Bot-BasicBot-Pluggable,
version 0.88:

  extended simple IRC bot for pluggable modules

This software is copyright (c) 2010 by Mario Domgoergen.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Perl extension Bot::BasicBot::Pluggable.

0.93 - Mon Jun 27 10:40:24 CDT 2011
 - when creating temporary files for Store::Storable->save, make sure
   they are created in the same directory that the final file will be
   moved to.  if not, the rename may fail if the source and destination
   are on separate filesystems.  (RT #68886)
 - correctly parse relative paths from *.storable file globbing in the
   Storable backend store.  this was preventing the store from restoring
   data for modules.  (RT #68887)

0.92 - Thu May  5 08:39:08 CDT 2011
 - added ability to load modules of arbitrary depth, courtesy of david
   precious (bigpresh)
 - refactor Auth module so that it doesn't step on other modules by
   eating all bang commands, also courtesy of bigpresh.
 - added Crypt::SaltedHash to the prereqs list

0.91 - Tue Dec  7 00:05:51 CST 2010
 - added transparent support for salted password hashes (RT #63232),
   courtesy of david precious (bigpresh)
 - fix busted flood protection (RT #62864), courtesy of david precious
   (bigpresh)
 - fix minor warning-producing exception in ChanOp->isop, also courtesy
   of the B.P.

0.90 - Fri Nov  5 17:14:30 CDT 2010

 - bumped dependency on Log::Log4perl from 0 to 1.11 due to dependency
   on the trace log level (RT #56540)

0.89 - Thu Oct 28 18:09:18 CDT 2010

  - Added DBD::SQLite and YAML::XS to the test dependency list
  - Quote table identifiers in Bot::BasicBot::Pluggable::Store::DBI
  - Bot::BasicBot::Pluggable is under new maintenance \o/

0.88 2010/08/29

  Removing build dependencies for Test::Perl::Critic and Test::Tidy.

  Bot::BasicBot::Pluggable is looking for a new maintainer  (again...)
  as the current maintainer is not using it anymore. Please mail
  to <mdom@cpan.org> if you are interested in taking over this fine
  module.

0.87 2010/05/09

  Remove all recommends due some failing tests. In a future release,
  some of the modules with long dependencies will be removed from
  this distribution and form their own on CPAN. I look at you DBI...

  Switched to Dist::Zilla

0.86 2010/03/28

  Thanks to Jacob Helwig for the follwoing changes:

    Join: Check for auth after checking whether we should handle
    the message at

    Add a new check to Bot::BasicBot::Pluggable::Module::Join to see whether
    the module should handle the message at all, and move the "is the user
    authed" after this check.

    By doing this we no longer send out "Sorry, ..." messages if someone
    happens to say "botname: Hi there!" without being authorized.

0.85 2010/02/13

  This is the David-Precious-aka-bigpresh-release, who commited all
  changes via Github. Thanks!

  - Join: Don't accept commands to join/part channels unless the
    user asking us to do so is auth'd to the bot
  - ChanOp: Make the ChanOp module actually work (uhm...)

0.84 2009/12/11

  - Log4Perl: Instead of just printing error messages to stdout,
    the bot now uses Log4perl to emit warnings and debugging information.
    For now i just added this new capability to Bot::BasicBot::Pluggable
    but this will be expanded to all the modules in one of the next
    releases.

    The bot's old log() function will now emit a message with
    priority warning to mimic the old behaviour.

    The command line bots now have two new options: --loglevel and
    --logconfig.

  - Commandline Bot: A simple command line bot was part of the
    github repository for quite some time now, but it was never
    shipped with a release before. To close #46875 and as it's a
    really neat way to test some modules, i'll reworked
    bot-basicbot-pluggable-cli, added
    App::Bot::BasicBot::Pluggable::Console and
    Bot::BasicBot::Pluggable::Console and use
    Test::Bot::BasicBot::Pluggable as its basis. You can call it
    with the same arguments as its irc cousin, but some options
    like --server or even --channel does not (yet?) make any sense
    in this context.

  - Auth: Added allow_anonymous as requested and implemented by
    Toni Spets. Setting this value to true, does basically disable
    authentication for all modules, that are not explicitly checking
    for the users authentication level.

    You only ever want to set this value to true, if you want to
    develop modules that need to prefix commands with ! for
    unauthenticated users. You've been warned.

  - Loader, Vars: As direct consequence Loader.pm and Vars.pm do
    now check explicitly.

  - Loader: !list does now also return all available but not loaded
    modules (thanks to Manuel Hachtkemper for the hint). I can see
    that the old behaviour might be a little bit confusing for some
    users.

  - available_modules() returns really all available modules ...
    till this release the bot just returned modules in PERL5LIB and
    ignored modules in ./modules and ./

  - Store: DBI, DBM::Deep and DBD::SQLite are no langer hard
    requirements of this module. Bot::BasicBot::Pluggable now tries
    to load all storage types in a predefined order until one returns
    successfully. Normally you should just select one explicitly.

  - Base.pm: Removed Base.pm from the list of available modules und
    depracate its usage

  - Try::Tiny: Error handling is now performed by try::tiny instead
    of writing all the eval boilerplate. Although it does lack some
    convenience, i selected it as we'll get it through moose in any
    case.

  - ChanOp.pm: New core module to handle channel managment operations
    like oping, deoping, kicking and flood control. Please take a
    closer look at this module, it definitly needs some testing!

0.83 2009/11/08

  - added four missing dispatch events from Bot::Basicbot: nick_change,
    kicked, topic and userquit. All four can now be handled by modules.
  - converted some evaled expressions to code blocks
  - documented a problem with hash signs in yaml configuration files

0.82 2009/08/31

  - Failing tests on cpantesters as two tests are creating intermediate

Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'DBD::SQLite' => '0',
    'FindBin' => '0',
    'Module::Build' => '0.3601',
    'Test::More' => '0',
    'YAML::XS' => '0'
  },
  'configure_requires' => {
    'ExtUtils::MakeMaker' => '6.30',
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'extended simple IRC bot for pluggable modules',
  'dist_author' => [
    'Mario Domgoergen <mdom@cpan.org>'
  ],
  'dist_name' => 'Bot-BasicBot-Pluggable',
  'dist_version' => '0.93',
  'license' => 'perl',
  'module_name' => 'Bot::BasicBot::Pluggable',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'Bot::BasicBot' => '0.60',
    'CGI' => '0',
    'Carp' => '0',
    'Config::Find' => '0',
    'Crypt::SaltedHash' => '0.06',
    'DBI' => '0',
    'DBM::Deep' => '0',
    'Data::Dumper' => '0',
    'File::Copy' => '0',
    'File::Spec' => '0',
    'File::Temp' => '0',
    'LWP::UserAgent' => '0',
    'List::MoreUtils' => '0',
    'Log::Log4perl' => '1.11',
    'Log::Log4perl::Level' => '0',
    'Module::Load' => '0',
    'Module::Pluggable' => '0',
    'Moose' => '0',
    'Moose::Util::TypeConstraints' => '0',
    'MooseX::Getopt::Dashes' => '0',
    'MooseX::SimpleConfig' => '0',
    'POE' => '0',
    'Socket' => '0',
    'Storable' => '0',
    'Text::Unidecode' => '0',
    'Try::Tiny' => '0',
    'URI' => '0',
    'URI::Find::Simple' => '0',
    'URI::Title' => '0',
    'XML::Feed' => '0',
    'perl' => '5.8.0'
  },
  'script_files' => [
    'bin/bot-basicbot-pluggable',
    'bin/bot-basicbot-pluggable-cgi',
    'bin/bot-basicbot-pluggable-cli',
    'bin/bot-basicbot-pluggable-infobot-upgrade',
    'bin/storable2dbi.pl'
  ]
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
